tests/zfs-tests/tests/functional/shared_log/shared_log_006_neg.ksh

```diff
@@ -0,0 +1,73 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or https://opensource.org/licenses/CDDL-1.0.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2023 by Delphix. All rights reserved.
+#
+
+. $STF_SUITE/tests/functional/shared_log/shared_log.kshlib
+
+#
+# DESCRIPTION:
+#	Negative shared log testing.
+#
+# STRATEGY:
+#	1. Create shared log pool & client
+#	2. Write some data to the client pool
+#	3. Scrub client and provider pools
```

Comment:
I think this should probably be:
1. Attempt to create a client pool with a missing shared log pool
2. Attempt to create a client pool with mis-named shared log pool
3. Attempt to create a client pool with a shared log and a log device
4. Attempt to use a client pool after the shared log has been destroyed
5. Attempt to create a client pool when the feature is disabled
6. Attempt to export/destroy an active shared log
7. Attempt to reguid a client/log pool
8. Attempt to checkpoint a client/log pool